test/integration/test_migration.py

```diff
@@ -0,0 +1,304 @@
+                        rule_description text,
+                        total_records integer,
+                        failed integer,
+                        median_30_day_failed double precision,
+                        failed_percentage double precision,
+                        passed integer,
+                        median_30_day_passed double precision,
+                        passed_percentage double precision,
+                        status text,
+                        time_filter text,
+                        task_ts timestamp with time zone not null,
+                        created_at timestamp with time zone default now() not null,
+                        id bigserial not null
+                            constraint {table_name}_pkey
+                                primary key,
+                        constraint {table_name}_unique_quality_check
+                            unique (attribute, rule_name, task_ts, time_filter)
+                    );
+                """
+
+
+class TestMigrationsResolverInit(unittest.TestCase):
+    def setUp(self):
+        """
+        Init a temporary table with some data.
+        """
+
+        sql = [
+            f"DROP SCHEMA IF EXISTS {DATA_QUALITY_SCHEMA} CASCADE;"
+            f"CREATE SCHEMA IF NOT EXISTS {DATA_QUALITY_SCHEMA};",
+            get_quality_table_creation_script(
+                DATA_QUALITY_SCHEMA, DATA_QUALITY_TABLE_1
+            ),
+            get_quality_table_creation_script(
+                DATA_QUALITY_SCHEMA, DATA_QUALITY_TABLE_2
+            ),
+        ]
+        self.conn = Connector(TEST_DB_URI)
+        for s in sql:
+            self.conn.execute(s)
+
+    def tearDown(self):
+        """
+        Drop all created tables.
+        """
+        self.conn.execute(f"DROP schema {DATA_QUALITY_SCHEMA} CASCADE;")
+        DQBase.metadata.clear()
+
+    def test_migration_table_exists_init(self):
+        versions_migrations = {"0.1.4": "54f8985b0ee5", "0.1.5": "480e6618700d"}
+
+        m = MigrationsResolver(
+            versions_migrations, "0.1.4", SQLALCHEMY_URL, DATA_QUALITY_SCHEMA
+        )
+        migration_table_exists = m.migrations_table_exists()
+
+        assert migration_table_exists is False
+
+    def test_get_current_migration_init(self):
+        versions_migrations = {"0.1.4": "54f8985b0ee5", "0.1.5": "480e6618700d"}
+
+        m = MigrationsResolver(
+            versions_migrations, "0.1.4", SQLALCHEMY_URL, DATA_QUALITY_SCHEMA
+        )
+        current = m.get_applied_migration()
+
+        assert current is None
+
+    def test_is_on_head_init(self):
+        versions_migrations = {"0.1.4": "54f8985b0ee5", "0.1.5": "480e6618700d"}
+
+        m = MigrationsResolver(
+            versions_migrations, "0.1.4", SQLALCHEMY_URL, DATA_QUALITY_SCHEMA
+        )
+        is_on_head = m.is_on_head()
+
+        is_on_head is False
+
+    def test_get_migrations_to_head__is_before_init(self):
+        versions_migrations = {
```

Comment:
You can define it as fixture or variable once and use it everywhere.

Reply:
Also format of the version hashes can be more readable, so it would be easier to follow what the test does.
`
{ "0.1.5": "0.1.5-hash"}
`

Reply:
in the tests there are a lot of versions of versions_migrations and in the case the same versions_migrations is used, there is very useful to have the code visually inside of tests to see what is tested

maybe version [ "0.1.5-hash", "0.1.6-hash"] would be more readable, but it would mean we must have additional code for the dictionary creation. While this is only internal thing (it is not exposed to user of the Contessa) I would prefer to use original way of map definition.

Reply:
Ad fixtures - ok, if there is too much variance, it would not simplify things :)
Ad hashes - I meant it only for test fixtures, but not changing format, only readable "hash strings", as I believe those are purely artificial in tests.

Reply:
ok, now I understand what did you mean :)